### DISCUSSION TITLE
When i use react error boundary + asyncAtom the resetErrorBoundary function not works

### DISCUSSION THREAD

**Discussion by alexrequelme** [pmndrs/jotai#2002] [Q&A]
I have been testing my endpoint simulating an error case to test how my frontend handles that error.
I have this atom that uses a promise to retrieve user favorites songs.
```js
export const asyncSongs = atom(getFavSongs);
```
My react tree is something like that
```js
<Provider>
  <ErrorBoundary FallbackComponent={InvalidUserComponent}>
    <Suspense>
      <SongsList />
    </Suspense>
  </ErrorBoundary>
</Provider>
```
I use the InvalidUserComponent to catch the error
```js
export default function InvalidUserComponent({ resetErrorBoundary }: FallbackProps) {
  return (
    <CenterScreen>
      <Text style={tw`font-semibold text-xl mt-6 mb-3`}>Something went wrong.</Text>
      <AppButton text="Try Again" onPress={resetErrorBoundary} />
    </CenterScreen>
  );
}
```
When an error is thrown, the ErrorBoundary works correctly by handling the error. However, when I click on 'try again,' which triggers my API to return a valid response, the atom doesn't fetch the songs. Based on my understanding, resetting the error boundary should cause the children to re-render, and the async atom should initiate a new fetch instead. Am I mistaken?

**Comment by dai-shi** [comment]
Atom values are stored in Provider. This means even if children re-renders, atoms are not re-evaluated. (Otherwise, there will be too many re-renders which is the reason to use the library.)

Atoms are only re-evaluated when their dependencies change. What I have done is something like [atomWithRefresh](https://jotai.org/docs/recipes/atom-creators#atomwithrefresh).

That said, practice with error boundary is still to be explored. Hope you come up with something new.
